api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -718,17 +718,20 @@ public String getSecretQuestion(User user) throws APIException {
     }
 	
 	/**
-	 * @see org.openmrs.api.UserService#getUserByEmail(java.lang.String)
-	 * 
+	 * @see org.openmrs.api.UserService#getUserByUsernameOrEmail(java.lang.String)
 	 */
 	@Override
 	@Transactional(readOnly = true)
-	public User getUserByEmail(String email) {
-		if (StringUtils.isNotBlank(email)) {
-			return dao.getUserByEmail(email);
+	public User getUserByUsernameOrEmail(String usernameOrEmail) {
+		if (StringUtils.isNotBlank(usernameOrEmail)) {
+			User user = dao.getUserByEmail(usernameOrEmail);
+			if (user == null) {
+				return getUserByUsername(usernameOrEmail);
+			} else {
```

Comment:
This else clause is redundant since the else clause always returns, just return the user